process/factory/metachain/vmContainerFactory.go

```diff
@@ -103,6 +106,11 @@ func NewVMContainerFactory(args ArgsNewVMContainerFactory) (*vmContainerFactory,
 	}
 	cryptoHook := hooks.NewVMCryptoHook()
 
+	appStatusHandler := args.StatusHandler
+	if appStatusHandler == nil {
```

Comment:
check.IfNil ?

Also, what happens if someone mistakenly stop passing the StatusHAndler to the arguments? This will silently create a new status handler and the metric won't show up. Suggestion to properly check for nil and return error.

Reply:
done